### PR TITLE
Revamp Orleans shopping cart example | .NET 8 + Azure Cosmos DB for NoSQL

### PR DESCRIPTION
**Pull request by seesharprun**
Changes:
- [x] Change package versions from static to floating (ex. `7.x`)
- [x] Pushed forward to .NET 8
- [x] Ran `dotnet format`
- [x] Updated Mud icons to fix deprecation warnings
- [x] Added null checking-ignore to connection string
- [x] Removed port metadata (not needed for sample app)
- [x] Revamped Program.cs to match the [Orleans quickstart](https://learn.microsoft.com/dotnet/orleans/quickstarts/deploy-scale-orleans-on-azure)
- [x] Implement Azure Cosmos DB for NoSQL grain persistence
- [x] Fix error where application is publishing to Port 8080 on Azure Container Apps
- [x] Add permanent Dockerfile and .dockerignore
- [x] Updated readme

Related to dotnet/docs#36975

### CODE REVIEW

**Comment by seesharprun** [comment]
@IEvangelist and @bradygaster, this sample will be ready for review next week after I add the Dockerfile

File: orleans/ShoppingCart/Grains/Orleans.ShoppingCart.Grains.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+    <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Orleans.Runtime" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Orleans.Sdk" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Orleans.Runtime" Version="7.*" />
+    <PackageReference Include="Microsoft.Orleans.Sdk" Version="7.*" />
```
Comment (IEvangelist) [inline orleans/ShoppingCart/Grains/Orleans.ShoppingCart.Grains.csproj:11]:
```suggestion
    <PackageReference Include="Microsoft.Orleans.Runtime" Version="8.*" />
    <PackageReference Include="Microsoft.Orleans.Sdk" Version="8.*" />
```

File: orleans/ShoppingCart/Abstractions/Orleans.ShoppingCart.Abstractions.csproj [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Orleans.Sdk" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Orleans.Sdk" Version="7.*" />
```
Comment (IEvangelist) [inline orleans/ShoppingCart/Abstractions/Orleans.ShoppingCart.Abstractions.csproj:10]: Hold for Orleans 8.0.0-rc.2

```suggestion
    <PackageReference Include="Microsoft.Orleans.Sdk" Version="8.*" />
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]

**Comment by seesharprun** [comment]
@IEvangelist, How do you feel about a [floating version resolution](https://learn.microsoft.com/nuget/concepts/package-versioning#floating-version-resolutions)?

```xml
<PackageReference Include="Microsoft.Orleans.Server" Version="8.*-*" />
```

That syntax should cover both pre-release and release variants of 8.x

I use this technique for [`Azure.AI.OpenAI`](https://www.nuget.org/packages/Azure.AI.OpenAI) over in our Cosmos+OpenAI training module: 

- https://github.com/Azure-Samples/cosmosdb-chatgpt/blob/main/cosmoschatgpt.csproj#L13

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**